identity/src/main/java/com/android/identity/internal/Util.kt

```diff
@@ -0,0 +1,2209 @@
+        octetStringSize: Int,
+        fieldValue: BigInteger?
+    ): ByteArray {
+        return BigIntegers.asUnsignedByteArray(octetStringSize, fieldValue)
+    }
+
+    @JvmStatic
+    fun cborBuildCoseKey(
+        key: PublicKey,
+        curve: EcCurve
+    ): DataItem {
+        val x: ByteArray
+        val y: ByteArray
+        val item: DataItem
+        when (curve) {
+            EcCurve.ED25519, EcCurve.ED448 -> {
+                x = (key as BCEdDSAPublicKey).pointEncoding
+                item = CborBuilder()
+                    .addMap()
+                    .put(COSE_KEY_KTY, COSE_KEY_TYPE_OKP)
+                    .put(COSE_KEY_PARAM_CRV, curve.coseCurveIdentifier.toLong())
+                    .put(COSE_KEY_PARAM_X, x)
+                    .end()
+                    .build().get(0)
+            }
+
+            EcCurve.X25519, EcCurve.X448 -> {
+                x = (key as BCXDHPublicKey).uEncoding
+                item = CborBuilder()
+                    .addMap()
+                    .put(COSE_KEY_KTY, COSE_KEY_TYPE_OKP)
+                    .put(COSE_KEY_PARAM_CRV, curve.coseCurveIdentifier.toLong())
+                    .put(COSE_KEY_PARAM_X, x)
+                    .end()
+                    .build().get(0)
+            }
+
+            else -> {
+                val ecKey = key as ECPublicKey
+                val w = ecKey.w
+                val keySizeBits: Int = curve.bitSize
+                x = sec1EncodeFieldElementAsOctetString((keySizeBits + 7) / 8, w.affineX)
+                y = sec1EncodeFieldElementAsOctetString((keySizeBits + 7) / 8, w.affineY)
+                item = CborBuilder()
+                    .addMap()
+                    .put(COSE_KEY_KTY, COSE_KEY_TYPE_EC2)
+                    .put(COSE_KEY_PARAM_CRV, curve.coseCurveIdentifier.toLong())
+                    .put(COSE_KEY_PARAM_X, x)
+                    .put(COSE_KEY_PARAM_Y, y)
+                    .end()
+                    .build().get(0)
+            }
+        }
+        return item
+    }
+
+    // Used for testing only, to create an invalid COSE_Key
+    @JvmStatic
+    fun cborBuildCoseKeyWithMalformedYPoint(key: PublicKey): DataItem {
+        val ecKey =
+            key as ECPublicKey
+        val w = ecKey.w
+        val malformedY = BigInteger.valueOf(42)
+        val x =
+            sec1EncodeFieldElementAsOctetString(32, w.affineX)
+        val y =
+            sec1EncodeFieldElementAsOctetString(32, malformedY)
+        return CborBuilder()
+            .addMap()
+            .put(COSE_KEY_KTY, COSE_KEY_TYPE_EC2)
+            .put(COSE_KEY_PARAM_CRV, COSE_KEY_PARAM_CRV_P256)
+            .put(COSE_KEY_PARAM_X, x)
+            .put(COSE_KEY_PARAM_Y, y)
+            .end()
+            .build().get(0)
+    }
+
+    @JvmStatic
+    fun cborMapHasKey(map: DataItem, key: String): Boolean {
+        val item = castTo<Map, DataItem>(
+            Map::class.java, map
+        )[UnicodeString(key)]
+        return item != null
```

Comment:
the `castTo` types are inferred, we can then use "equals" instead of `return`

```
    fun cborMapHasKey(map: DataItem, key: String): Boolean =
        castTo(
            Map::class.java, map
        )[UnicodeString(key)] != null
```